src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java

```diff
@@ -154,7 +158,7 @@ private List<ResultTransformerConfiguration> getResultTransformerConfigurations(
         }
 
         // Fetch all index settings for this plugin
-        String[] settingNames = supportedResultTransformers.values()
+        String[] settingNames = resultTransformerMap.values()
                 .stream()
                 .map(t -> t.getTransformerSettings()
                         .stream()
```

Comment:
I just realized that this can be simplified down to one `flatMap` like:

```
        String[] settingNames = resultTransformerMap.values()
                .stream()
                .flatMap(t -> t.getTransformerSettings()
                        .stream()
                        .map(Setting::getKey))
                .toArray(String[]::new);
```